packages/utils/src/conversions.ts

```diff
@@ -0,0 +1,202 @@
+/**
+ * @module Conversions
+ * This module provides a collection of utility functions for converting
+ * between different numerical formats, particularly focusing on
+ * conversions involving bigints, hexadecimals and buffers.
+ * The module is structured with clear function naming to indicate
+ * the conversion direction (e.g., `bigIntToHexadecimal` for BigInt
+ * to hexadecimal, `bufferToBigint` for buffer to bigint) and employs
+ * type checks to ensure the correct handling of various input types.
+ * It also includes variations for both big-endian (`be`) and little-endian
+ * (`le`) conversions. It is important to note that when there is no prefix,
+ * the order of bytes is always big-endian.
+ */
+
+import { BigNumberish } from "./types"
+import { isHexadecimal, isStringifiedBigint } from "./number-checks"
+
+/**
+ * Converts a bigint to a hexadecimal string ensuring even length.
+ * @param n The bigint value to convert.
+ * @returns The hexadecimal representation of the bigint.
+ */
+export function beBigIntToHexadecimal(n: bigint): string {
+    let hex = n.toString(16)
+
+    // Ensure even length.
+    if (hex.length % 2 !== 0) {
+        hex = `0${hex}`
+    }
+
+    return hex
+}
+
+/**
+ * Converts a bigint to a hexadecimal string. Alias for beBigIntToHexadecimal.
+ * @param n The bigint value to convert.
+ * @returns The hexadecimal representation of the bigint.
+ */
+export function bigIntToHexadecimal(n: bigint): string {
```

Comment:
For hexadecimal stringification of a number, I actually don't think byte order needs to be explicit.  It doesn't hurt, but I think in this case there actually is an obvious default.  Numbers represented as strings always put the most significant digit first.  Importantly these functions are really outputting digits (4 bits at a time) not bytes.

Relatedly, in this case it's weird to make BE byte order explicit, but not provide any LE alternatives.

Reply:
I see. I'd tend to prefer leaving it without any prefix then. If we realize someone needs this conversion with little-endian byte order we could make it more explicit. What do you think?

Reply:
I agree.